### PR TITLE
Spring 06

### PR DESCRIPTION
**Pull request by antoniukivan**

### CODE REVIEW

File: src/main/java/ua/com/cinema/controllers/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    @PostMapping("/complete")
+    public void complete(@RequestParam Long userId) {
+        User user = userService.getById(userId).get();
+        orderService.completeOrder(shoppingCartService.getByUser(user));
+    }
+
+    @GetMapping
+    public List<Order> getOrdersHistory(@RequestParam Long userId) {
+        return orderService.getOrdersHistory(userService.getById(userId).get());
```
Comment (yurwing) [inline src/main/java/ua/com/cinema/controllers/OrderController.java]: you don't have to return entity
```suggestion
    @GetMapping
    public List<OrderResponseDto> getOrdersHistory(@RequestParam Long userId) {
        return orderService.getOrdersHistory(userService.getById(userId).get());
```

File: src/main/java/ua/com/cinema/service/mappers/OrderMapper.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Override
+    public OrderResponseDto getDtoFromModel(Order order) {
+        OrderResponseDto orderResponseDto = new OrderResponseDto();
+        orderResponseDto.setId(order.getId());
+        orderResponseDto.setPurchaseTime(order.getPurchaseTime().toString());
+        List<Long> ticketsId = ticketService.getAllByUser(order.getUser()).stream()
+                .map(Ticket::getId)
+                .collect(Collectors.toList());
+        orderResponseDto.setTickets(ticketsId);
```
Comment (yurwing) [inline src/main/java/ua/com/cinema/service/mappers/OrderMapper.java]: here you can use order.getTickets() instead of using ticketService.getAllByUser()

**Review by yurwing — CHANGES_REQUESTED** [review]

**Review by yurwing — APPROVED** [review]

File: src/main/java/ua/com/cinema/dao/AbstractDao.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+public abstract class AbstractDao<T, I extends Serializable> {
+    private final SessionFactory sessionFactory;
+    private final Class<T> clazz;
+
+    protected AbstractDao(SessionFactory sessionFactory, Class<T> clazz) {
+        this.sessionFactory = sessionFactory;
+        this.clazz = clazz;
+    }
+
+    public SessionFactory getSessionFactory() {
```
Comment (boroda4436) [inline src/main/java/ua/com/cinema/dao/AbstractDao.java]: What about a `protected` access modifier?

File: src/main/java/ua/com/cinema/dao/AbstractDao.java [hunk 4]
```diff
@@ -17,14 +17,10 @@ protected AbstractDao(SessionFactory sessionFactory, Class<T> clazz) {
         this.clazz = clazz;
     }
 
-    public SessionFactory getSessionFactory() {
+    protected SessionFactory getSessionFactory() {
```
Comment (boroda4436) [inline src/main/java/ua/com/cinema/dao/AbstractDao.java:20]: You can make your field protected and remove this getter

**Review by boroda4436 — APPROVED** [review]

File: src/main/java/ua/com/cinema/dao/AbstractDao.java (hunk 4 above)
Comment (antoniukivan) [reply src/main/java/ua/com/cinema/dao/AbstractDao.java:20]: 😢

**Pull request merged**